File: android/app/src/main/java/com/dibay/app/DibayAppIconDeliveryPlugin.java
package com.dibay.app;

import com.getcapacitor.JSObject;
import com.getcapacitor.Plugin;
import com.getcapacitor.PluginCall;
import com.getcapacitor.PluginMethod;
import com.getcapacitor.annotation.CapacitorPlugin;

/**
 * Capacitor bridge for Android Delivery Adapter v1.
 * JS passes projected appIconTotal only — no Kernel recalculation.
 */
@CapacitorPlugin(name = "DibayAppIconDelivery")
public class DibayAppIconDeliveryPlugin extends Plugin {

  @PluginMethod
  public void apply(PluginCall call) {
    Integer count = call.getInt("count");
    if (count == null) {
      call.reject("count_required");
      return;
    }
    DibayAppIconDeliveryAdapter.apply(getContext(), count);
    JSObject ret = new JSObject();
    ret.put("ok", true);
    ret.put("count", Math.max(0, count));
    call.resolve(ret);
  }
}

File: ios/App/App/Plugins/DibayAppIconDeliveryPlugin.swift
import Capacitor
import Foundation
import os.log

/**
 * Capacitor bridge for iOS Delivery Adapter v1.
 * JS passes projected appIconTotal only — no Kernel recalculation.
 *
 * CAPBridgedPlugin — must stay in packageClassList (see patch-ios-capacitor-package-class-list.mjs).
 */
@objc(DibayAppIconDeliveryPlugin)
public class DibayAppIconDeliveryPlugin: CAPPlugin, CAPBridgedPlugin {
  private static let log = OSLog(subsystem: "com.dibay.app", category: "DIBAY_APPICON_DELIVERY")

  public let identifier = "DibayAppIconDeliveryPlugin"
  public let jsName = "DibayAppIconDelivery"
  public let pluginMethods: [CAPPluginMethod] = [
    CAPPluginMethod(name: "apply", returnType: CAPPluginReturnPromise),
  ]

  @objc func apply(_ call: CAPPluginCall) {
    guard let count = call.getInt("count") else {
      call.reject("count_required")
      return
    }
    DibayAppIconDeliveryAdapter.apply(appIconTotal: count)
    os_log("plugin_apply total=%{public}d", log: Self.log, type: .info, count)
    call.resolve([
      "ok": true,
      "count": max(0, count),
    ])
  }
}
